### PR TITLE
Functional version of evcurve workflow

### PR DESCRIPTION
**Pull request by jan-janssen**

### CODE REVIEW

File: atomistics/workflows/evcurve/helper.py [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+        volume_lst=volume_lst,
+        energy_lst=energy_lst,
+        fit_type=fit_type,
+        fit_order=fit_order,
+    ).fit_dict
+
+
+def get_strains(
+    vol_range: Optional[float],
+    num_points: Optional[int],
```
Comment (liamhuber) [inline atomistics/workflows/evcurve/helper.py:103]: Doesn't look optional

File: tests/test_elastic_lammps_functional.py [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+                [0., 0., 0., 51.23853765, 0., 0.],
+                [0., 0., 0., 0., 51.23853765, 0.],
+                [0., 0., 0., 0., 0., 51.23853765]
+            ])
+        ).all())
+        self.assertEqual(sym_dict['SGN'], 225)
+        self.assertEqual(sym_dict['LC'], 'CI')
+        self.assertEqual(sym_dict['Lag_strain_list'], ['01', '08', '23'])
+        self.assertTrue(np.isclose(sym_dict['epss'], np.array([-0.005, -0.0025, 0., 0.0025, 0.005])).all())
+        self.assertTrue(np.isclose(sym_dict["v0"], 66.43035441556098))
```
Comment (liamhuber) [inline tests/test_elastic_lammps_functional.py:67]: If the dict value is as scalar as it looks you can cut numpy and directly use `assertAlmostEqual`

Comment (jan-janssen) [reply tests/test_elastic_lammps_functional.py]: Fixed, I also applied it to the other tests in https://github.com/pyiron/atomistics/pull/251

File: atomistics/workflows/evcurve/helper.py (hunk 1 above)
Comment (jan-janssen) [reply atomistics/workflows/evcurve/helper.py:103]: It is either `vol_range` and `num_points` or `strain_lst`.

Comment (liamhuber) [reply atomistics/workflows/evcurve/helper.py:103]: It categorically is not. The code as written does not allow `vol_range` and `num_points` to be optional. It is `vol_range` and `num_points`, or `vol_range` and `num_points` (both ignored) and `strain_list`. 

```python
>>> from atomistics.workflows.evcurve.helper import get_strains
>>> 
>>> get_strains(strain_lst=[1,2,3])
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
Cell In[6], line 3
      1 from atomistics.workflows.evcurve.helper import get_strains
----> 3 get_strains(strain_lst=[1,2,3])

TypeError: get_strains() missing 2 required positional arguments: 'vol_range' and 'num_points'
```

You've placed two purely positional arguments with no default and then hinted them as optional.

Comment (jan-janssen) [reply atomistics/workflows/evcurve/helper.py:103]: This is now addressed in https://github.com/pyiron/atomistics/pull/252

**Pull request merged**